src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -55,6 +55,18 @@ def _attach_related(_events):
             event.group = groups.get(event.group_id)
         eventstore.bind_nodes(_events, "data")
 
+    def _attach_timestamp(_events):
+        """
+        When migrating old data from Sentry 9.0.0 to 9.1.2 to 10 in rapid succession, the event timestamp may be missing.
+        This function exists to attach the timestamp if its missing
+        """
+        for event in _events:
+            if not event.data.data.get("timestamp"):
+                # Python 2.7 way to convert datetime to timestamp.
+                timestamp = ((e.datetime.replace(tzinfo=None) - e.datetime.utcoffset()) - datetime(1970, 1, 1)).total_seconds()
+                e.data.data['timestamp'] = timestamp
```

Comment:
Don't think `e` is defined here?

Reply:
Ah you're right. I had forgotten to update this after I moved it into the callback